controllers/applicationsnapshotenvironmentbinding_controller.go

```diff
@@ -225,13 +225,35 @@ func (r *SnapshotEnvironmentBindingReconciler) Reconcile(ctx context.Context, re
 			}
 		}
 
+		// Generate a route name for the component
+
 		kubernetesResources, err := devfile.GetResourceFromDevfile(log, compDevfileData, deployAssociatedComponents, hasComponent.Name, hasComponent.Spec.Application, hasComponent.Spec.ContainerImage, hostname)
 		if err != nil {
 			log.Error(err, "unable to get kubernetes resources from the devfile outerloop components")
 			r.SetConditionAndUpdateCR(ctx, req, &appSnapshotEnvBinding, err)
 			return ctrl.Result{}, err
 		}
 
+		// Create a unique, generated name for the route
+		// ToDo: Ideally we wouldn't need to loop here, but since the Component status is a list, we can't avoid it
```

Comment:
Should we consider making the BindingComponent status list a map (`map[componentName]BindingComponent`) instead? In a separate PR obviously 😄